modules/options/src/main/scala/scala/build/actionable/ActionableDependencyHandler.scala

```diff
@@ -28,62 +27,70 @@ case object ActionableDependencyHandler
 
   override def actionableDiagnostic(
     setting: Positioned[AnyDependency],
-    buildOptions: BuildOptions
+    buildOptions: BuildOptions,
+    loggerOpt: Option[Logger]
   ): Either[BuildException, Option[ActionableDependencyUpdateDiagnostic]] = either {
-    val dependency     = setting.value
-    val currentVersion = dependency.version
-    val latestVersion  = value(findLatestVersion(buildOptions, dependency))
-    if Version(latestVersion) > Version(currentVersion) && !isLatestSyntaxVersion(currentVersion)
-    then
-      if dependency.userParams.contains(Constants.toolkitName) &&
-        dependency.module.name != Constants.toolkitTestName
+    val dependency       = setting.value
+    val currentVersion   = dependency.version
+    val latestVersionOpt = value(findLatestVersion(buildOptions, setting, loggerOpt))
+
+    for {
+      latestVersion <- latestVersionOpt
+      if Version(latestVersion) > Version(currentVersion) &&
+      !isLatestSyntaxVersion(currentVersion)
+      // filtering out toolkit-test to prevent double-update-diagnostic
+      if !(dependency.userParams.contains(Constants.toolkitName) &&
+      dependency.module.name == Constants.toolkitTestName)
+    } yield
+      if dependency.userParams.contains(Constants.toolkitName)
       then
         val toolkitSuggestion =
           if dependency.module.organization == Constants.toolkitOrganization then latestVersion
           else if dependency.module.organization == Constants.typelevelOrganization then
             s"typelevel:$latestVersion"
           else s"${dependency.module.organization}:$latestVersion"
-        Some(ActionableDependencyUpdateDiagnostic(
+        ActionableDependencyUpdateDiagnostic(
           setting.positions,
           currentVersion,
           latestVersion,
           dependencyModuleName = Constants.toolkitName,
           suggestion = toolkitSuggestion
-        ))
-      else if dependency.userParams.contains(Constants.toolkitName) &&
-        dependency.module.name == Constants.toolkitTestName
-      then None // filtering out toolkit-test to prevent double-update-diagnostic
+        )
       else
-        Some(ActionableDependencyUpdateDiagnostic(
+        ActionableDependencyUpdateDiagnostic(
           setting.positions,
           currentVersion,
           latestVersion,
           dependencyModuleName = dependency.module.name,
           suggestion = dependency.copy(version = latestVersion).render
-        ))
-    else
-      None
+        )
   }
 
   /** Versions like 'latest.*': 'latest.release', 'latest.integration', 'latest.stable'
     */
   private def isLatestSyntaxVersion(version: String): Boolean = Latest(version).nonEmpty
   private def findLatestVersion(
     buildOptions: BuildOptions,
-    dependency: AnyDependency
-  ): Either[BuildException, String] = either {
+    setting: Positioned[AnyDependency],
+    loggerOpt: Option[Logger]
+  ): Either[BuildException, Option[String]] = either {
```

Comment:
```suggestion
  ): Option[String]  = 
```

Reply:
Still some operations can fail in this function, so the Either is needed.